core_lib/src/graphics/bitmap/bitmapimage.cpp

```diff
@@ -808,6 +809,15 @@ void BitmapImage::floodFill(BitmapImage* targetImage,
     QRgb oldColor = targetImage->pixel(point);
     oldColor = qRgba(qRed(oldColor), qGreen(oldColor), qBlue(oldColor), qAlpha(oldColor));
 
+    QRgb fillColor = newColor;
+    if (fillMode == 1)
+    {
+        QColor tempColor;
+        tempColor.setRgba(newColor);
+        tempColor.setAlphaF(1);
+        fillColor = tempColor.rgba();
+    }
+
```

Comment:
I'd prefer if we could keep the floodfill algorithm as clean as possible. Couldn't this be moved to bucketTool instead?

Reply:
Done